sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1249,7 +1249,7 @@ class CodegenContext extends Logging {
       // at least two nodes) as the cost of doing it is expected to be low.
 
       val subExprCode = s"${addNewFunction(fnName, fn)}($INPUT_ROW);"
-      subexprFunctions += s"${addNewFunction(fnName, fn)}($INPUT_ROW);"
```

Comment:
oh, I see...

Reply:
`addNewFunction` will overwrite in the method map. So this won't add new function. But it is better to avoid duplicatie call.